extensions/2.0/Vendor/GRIFFEL_bim_data/README.md

```diff
@@ -0,0 +1,221 @@
+* ### Embed properties into glTF json
+```javascript
+"nodes": [
+  {
+    "name": "Door 1",
+    "extensions": {
+      "GRIFFEL_bim_data": {
+        "properties": [
+          0
+        ],
+        "type": 0
+      }
+    }
+  },
+  {
+    "name": "Door 2",
+    "extensions": {
+      "GRIFFEL_bim_data": {
+        "properties": [
+          3
+        ],
+        "type": 0
+      }
+    }
+  }
+],
+"extensions": [
+  "GRIFFEL_bim_data": {
+    "propertyNames": [
+      "Height",
+      "Width",
+      "Material"
+    ],
+    "propertyValues": [
+      "2,1 m",
+      "900 mm",
+      "Timber"
+      "2,4 m",
+    ],
+    "properties": [
+      {
+        "name": 0,
+        "value": 0
+      },
+      {
+        "name": 1,
+        "value": 1
+      },
+      {
+        "name": 2,
+        "value": 2
+      },
+      {
+        "name": 0,
+        "value": 3
+      }
+    ],
+    "types": [
+      {
+        "properties": [
+          1,
+          2
+        ]
+      }
+    ]
+  }
+]
+```
+
+* ### Write properties to separate binary file
+
+In this case each node points to the buffer view which references the buffer where metadata for this node can be found. File is a serialized 'GRIFFEL_bim_data' extension object with additional `nodeProperties` field. This field maps node properties and their types to nodes by node ID.
+
+[MessagePaсk](https://msgpack.org/) must be used for serializing extension object in a separate binary file to achive the most compact file size and the fastest deserialization speed.
+
+```javascript
+// gltf content:
+"nodes": [
+  {
+    "name": "Door 1",
+    "extensions": {
+      "GRIFFEL_bim_data": {
+        "bufferView": 99
+      }
+    }
+  },
+  {
+    "name": "Door 2",
+    "extensions": {
+      "GRIFFEL_bim_data": {
+        "bufferView": 99
+      }
+    }
+  }
+]
+
+"bufferViews": [
+  // buffer view indexed 99:
+  {
+    "buffer": 1,
+    "byteOffset": 0,
+    "byteLength": 1234
+  }
+  
+"buffers": [
+  // buffer indexed 1:
+  {
+    "uri": "projectName.meta",
+    "byteLength": 1234
+  }
+]
+
+// Content of 'projectName.meta':
+{
+  "nodeProperties": [
+    {
+      "node": 0,
+      "properties": [
+          0
+      ],
+      "type": 0
+    },
+    {
+      "node": 1,
+      "properties": [
+          3
+      ],
+      "type": 0
+    }
+  ],
+  // propertyNames, propertyValues, properties and types collections are the same
+}
+```
+
+So, for each node, if GRIFFEL_bim_data extension has `properties` or `type` - look for them in the same gltf json file. If the extension states `bufferView` - look for the data in the separate .meta file by the url from the buffer.
+Nodes can have both embedded and separately stored properties, in that case both are used. Exporters should take care of possible data duplication. Importers should treat embedded values as of higher priority than values, stored in a separate file, as the file may be obsolete or corrupted more likely.
```

Comment:
Consider adding that internal references of embedded and separate payloads have local scope, e.g., `type` reference from the embedded JSON refers to an element of the embedded `types` array.

Reply:
For simplicity I would say cross references between embedded and external data should not be allowed. Writing this down in _Readme.md_.